### ISSUE TITLE
Incorrect "Add 'this.'" diagnostic for class initializers

### ISSUE THREAD

**Issue by dpoeschl** [dotnet/roslyn#22776]
This is reported in a few Developer Community feedback items, I will link them as they're processed.
-  https://developercommunity.visualstudio.com/content/problem/131432/add-this-or-me-qualification-for-class-initializer.html
- https://developercommunity.visualstudio.com/content/problem/129705/incorrect-ide0009-in-object-initializer.html
- https://developercommunity.visualstudio.com/content/problem/161932/ide0009-add-this-or-me-qualification-incorrectly-r.html

I have not been able to reproduce this locally using the settings in the Tools | Options. Maybe it requires the editorconfig? That doesn't make much sense but I haven't been able to come up with anything else.

**Comment by mazhou-msft** [comment]
Repro on d15rel 27017.1 (15.5.0 Preview 2)

Create a new .NET Core Console App and then create a class such as:

```C#
public class A
{
    public string S { get; set; }
}
```

then in an **instance** method, use an initializer:

```C#
A a = new A
{
    S = string.Empty
}
```

Note that the brackets and the code must be in different lines to see the IDE0009 warning. Regardless where you set "preferred this"  option (either editorconfig or Tools -> Options.)

**Comment by paulvanbrenk** [comment]
Consistently see this in all object initializers in the Node Tools code base.

![image](https://user-images.githubusercontent.com/5273975/31918828-83d8a2d8-b813-11e7-9da7-e52051fc4e35.png)

**Comment by gundermanc** [comment]
@dpoeschl I have a 100% consistent repro in the VS editor code base as well. I suspect that you're right and this is based on the .editorconfig we're using.

**Comment by AArnott** [comment]
My repro is also in a repro that has [an .editorconfig file](https://github.com/AArnott/Nerdbank.GitVersioning/blob/master/src/.editorconfig).

**Comment by zaytsev-victor** [comment]
@sharwell @Pilchie Since it is very annoying regression is there a chance that it will be fixed in 15.5?
It is will be a blocker for any team which use 'Qualify member access' rule.

**Comment by Pilchie** [comment]
Looks like https://github.com/dotnet/roslyn/issues/23381 is a dupe of this.
